python/GafferSceneUI/RenderPassEditor.py

```diff
@@ -0,0 +1,750 @@
+				],
+				selectionMode = GafferUI.PathListingWidget.SelectionMode.Cells,
+				displayMode = GafferUI.PathListingWidget.DisplayMode.Tree,
+				horizontalScrollMode = GafferUI.ScrollMode.Automatic
+			)
+
+			self.__pathListing.buttonDoubleClickSignal().connectFront( Gaffer.WeakMethod( self.__buttonDoubleClick ), scoped = False )
+			self.__pathListing.keyPressSignal().connect( Gaffer.WeakMethod( self.__keyPress ), scoped = False )
+			self.__pathListing.buttonPressSignal().connectFront( Gaffer.WeakMethod( self.__buttonPress ), scoped = False )
+
+			self.__settingsNode.plugSetSignal().connect( Gaffer.WeakMethod( self.__settingsPlugSet ), scoped = False )
+
+		self._updateFromSet()
+		self.__updateColumns()
+
+	__columnRegistry = collections.OrderedDict()
+
+	@classmethod
+	def registerOption( cls, groupKey, optionName, section = None, columnName = None ) :
+
+		if not columnName :
+			columnName = optionName.split( ":" )[-1]
+
+		GafferSceneUI.RenderPassEditor.registerColumn(
+			groupKey,
+			optionName,
+			lambda scene, editScope : _GafferSceneUI._RenderPassEditor._RenderPassEditorInspectorColumn(
+				GafferSceneUI.Private.OptionInspector( scene, editScope, optionName ),
+				columnName
+			),
+			section
+		)
+
+	# Registers a column in the Render Pass Editor.
+	# `inspectorFunction` is a callable object of the form
+	# `inspectorFunction( scene, editScope )` returning a
+	# `GafferSceneUI._RenderPassEditor._RenderPassEditorInspectorColumn` object.
+	@classmethod
+	def registerColumn( cls, groupKey, columnKey, inspectorFunction, section = None ) :
+
+		sections = cls.__columnRegistry.setdefault( groupKey, collections.OrderedDict() )
+		section = sections.setdefault( section, collections.OrderedDict() )
+
+		section[columnKey] = inspectorFunction
+
+	def __repr__( self ) :
+
+		return "GafferSceneUI.RenderPassEditor( scriptNode )"
+
+	def __firstValidScenePlug( self, node ):
+
+		for plug in GafferScene.ScenePlug.RecursiveOutputRange( node ) :
+			if not plug.getName().startswith( "__" ):
+				return plug
+		return None
+
+	def _updateFromSet( self ) :
+
+		GafferUI.NodeSetEditor._updateFromSet( self )
+
+		# Decide what plug we're viewing.
+		self.__plug = None
+		self.__plugParentChangedConnection = None
+		self.__plugDirtiedConnections = []
+		node = self._lastAddedNode()
+		if node is not None :
+			self.__plug = self.__firstValidScenePlug( node )
+			if self.__plug is not None :
+				self.__plugParentChangedConnection = self.__plug.parentChangedSignal().connect(
+					Gaffer.WeakMethod( self.__plugParentChanged ), scoped = True
+				)
+				self.__plugDirtiedConnections.append( node.plugDirtiedSignal().connect( Gaffer.WeakMethod( self.__plugDirtied ), scoped = True ) )
+
+		self.__settingsNode["in"].setInput( self.__plug )
+
+		# call base class update - this will trigger a call to _titleFormat(),
+		# hence the need for already figuring out the plug.
+		GafferUI.NodeSetEditor._updateFromSet( self )
+
+		self.__setPathListingPath()
```

Comment:
I think this should be unnecessary, but possibly only in Gaffer 1.4 - see comments in `__setPathListingPath`.

Reply:
Yeah, it does seem unnecessary but requires 1.4. I've added todo to remove this in a7729fb